tests/nim.cfg

```diff
@@ -1 +1,7 @@
---threads:on
```

Comment:
Why only tests had this option turned on?

Reply:
we generally do not want libraries to impose their settings on the application in which they're used, so we have to be a little careful in what we add to cfg files - so as a general principle, application-wide flags should ideally be set in the "end" application - tests are "end" applications using/stressing the library

Reply:
so - I was under the impression that this would be loaded every time the library is used, but it only happens when projects are nested within each other - from what I remember about threads, many of our core libs should be tested both with threads on and off to avoid surprises - but for applications, we almost always want threads:on (generally the language turns more strict then)

Reply:
Does testing with threads:off make sense? I think having threads:on just enables more check (like gcsafe)

Reply:
we do it for some libraries because there are compile-time differences that are difficult to anticipate - ie you hit different bugs in the compiler more or less, if you have threads on and off

Reply:
but yeah, if we pick one option to test, it's threads:on, mainly because like you say, it's more strict (and less tested in nim in general, meaning if it works, it's usually the case that off works too)